code/modules/mob/honking_angel.dm

```diff
@@ -347,6 +347,16 @@
 	set name = "Power Disruption"
 	set desc = "Disrupt and drain power from the local area."
 	set category = "H.Angel"
+	//find current area
+	var/area/A = get_area(src.loc)
+	//A = A.loc
+	if (!( istype(A, /area))) // If its not a area, stop!
+		return
+	if(!A.requires_power) // If it doesn't require power, stop!
+		return
+	if (istype(A, /area/space)) // If it is space, LOVE OF GOD PLEASE STOP!
```

Comment:
I understand it's probably not you who coded it, but hardcoding is bad. There should be a variable that says that there isn't any power in the area, or that it doesn't need power. Should check for those. (Unless it's an issue, you don't need to change it.)